### PR TITLE
Comments: When post is missing a title use excerpt

### LINKED ISSUE #15455: Comments: when post is missing a title use excerpt or alt text
This is a follow up to https://github.com/Automattic/wp-calypso/pull/15315#discussion_r123341546

Sometimes posts do not have titles. Instead of displaying `untitled`, let's try to get a short except of the post content. If the post only contains an image, try to use it's alt text. If a post has no text or alt text, fallback to `untitled`

### PR DESCRIPTION
**Pull request by Copons**
Closes #15455 (M4 Task)

Currently missing post titles are simply replaced by "Untitled", as we don't necessarily have any other information about the post loaded in state.

This PR adds a `QueryPosts` component that only requests the post if:
- The post title included in the comment object is missing.
- And the post is not yet loaded in state.

Then, uses the post excerpt instead of the title.

| Before | After |
| --- | --- |
| screen shot 2017-10-16 at 17 27 45 | screen shot 2017-10-16 at 17 27 52<br /><br />screen shot 2017-10-16 at 17 27 59 |

### CODE REVIEW

**Comment by dmsnell** [comment]
Nice work! I'd encourage you to consider writing a function to get the post's _display title_ and base the query component's display off of that instead of tying it to the `post_title`. Why? the excerpt may also be blank.

In the notifications we use: `post_title`, first 100 words of post, then a fallback string if still empty.

**Comment by Copons** [comment]
@dmsnell Thanks! 🙇 

The idea is the same as for notes, but explicitly using the excerpt instead of the first 100 words of a post.

`isPostTitleLoaded: !! postTitle || !! post` protects exactly for that case—unless my brain can't see it. :)

The fallback string "Untitled" is used directly in the components, so that we can rely on a falsy `postTitle` in that check.

File: client/blocks/comment-detail/index.jsx [hunk 1]
```diff
@@ -400,8 +404,11 @@ const mapStateToProps = ( state, ownProps ) => {
 
 	const postId = get( comment, 'post.ID' );
 
-	// TODO: eventually it will be returned already decoded from the data layer.
-	const postTitle = decodeEntities( get( comment, 'post.title' ) );
+	const post = getSitePost( state, siteId, postId );
+
+	const postTitle =
+		decodeEntities( get( comment, 'post.title' ) ) ||
+		decodeEntities( stripHTML( get( post, 'excerpt' ) ) );
```
Comment (dmsnell) [inline client/blocks/comment-detail/index.jsx]: not a big deal, but if we have unicode whitespace characters we'll still get a visual blank. the RegExp `\s` though I believe takes those into consideration, so if we wanted to iterate…

```js
const getTitle = ( post, comment ) => {
	const ws = /^\s*$/;
	const title = decodeEntities( get( comment, 'post.title' ) );
	if ( ! ws.test( title ) ) {
		return title;
	}

	// might `post` be `undefined`?
	// if not, we can just use `post.excerpt`
	// and leave out the `get()`
	const excerpt = stripHTML( get( post, 'excerpt' ) );
	if ( ws.test( excerpt ) ) {
		return excerpt;
	}

	return translate( 'Untitled Post' );
}
```

Comment (Copons) [reply client/blocks/comment-detail/index.jsx]: Good catch, I'll add an issue about this.

**Pull request merged**